src/views/components/student-panel.jsx

```diff
@@ -21,19 +23,45 @@ export default class StudentPanel extends ListenerComponent {
             models: ModelStore.getAllModels()
         }
     }
+    renderDueDate(item){
+       let timeString = Moment.tz(item.timestamp * 1000, 'America/Chicago').format('MMMM Do YYYY, h:mm:ss a')
 
+        return (
+            <div>Due date: {timeString}</div>
+        )
+
+    }
+    showDateModal(){
+        this.setState({
+            showDateModal: true
+        });
+    }
+
+    changeDueDate(date, item){
+        this.setState({
+            showDateModal: false
+        });
+        StudentPanelController.changeDueDate(date, item);
+    }
     renderEquipment() {
         if(this.props.student.items.length === 0) {
             return <i className='equipment-none'>Student has no equipment checked out.</i>;
         }
+
         return (
             <div className='equipment'>
                 {this.props.student.items.map((item, i) => {
-                    return <Link to={`/item/${item.address}`}  key={i} className={item.timestamp < Math.floor(Date.now()/1000) ? 'link-nostyle overdue' : 'link-nostyle'}>
-                        <div className="item-info">
+                    return <div className="item-info"  key={i}>
+                        <DateModal
+                            message='Select new due date'
+                            active={this.state.showDateModal}
+                            onDateSelected={date => this.changeDueDate(date, item)}
+                        />
+                        <Link to={`/item/${item.address}`} className={item.timestamp < Math.floor(Date.now()/1000) ? 'link-nostyle overdue' : 'link-nostyle'}>
                             {this.renderItemInfo(item)}
+                        </Link>
+                        <div title="Click to change due date" onClick={this.showDateModal.bind(this, item)}>{this.renderDueDate(item)}</div>
```

Comment:
Nit: Why wrap it in another `<div>`? Just put the `title=` and `onClick` on the `div` in `renderDueDate`.